src/OpenTelemetry.Sampler.AWS/AWSXRayRemoteSampler.cs

```diff
@@ -43,20 +50,41 @@ internal AWSXRayRemoteSampler(TimeSpan pollingInterval, string endpoint)
 
     internal Timer RulePollerTimer { get; }
 
+    private RulesCache RulesCache { get; }
+
+    private Trace.Sampler FallbackSampler { get; }
+
     /// <summary>
     /// Initializes a <see cref="AWSXRayRemoteSamplerBuilder"/> for the sampler.
     /// </summary>
+    /// <param name="resource">an instance of <see cref="Resources.Resource"/>
+    /// to identify the service attributes for sampling. This resource should
+    /// be the same as what the OpenTelemetry SDK is configured with.</param>
     /// <returns>an instance of <see cref="AWSXRayRemoteSamplerBuilder"/>.</returns>
-    public static AWSXRayRemoteSamplerBuilder Builder()
+    public static AWSXRayRemoteSamplerBuilder Builder(Resource resource)
     {
-        return new AWSXRayRemoteSamplerBuilder();
+        return new AWSXRayRemoteSamplerBuilder(resource);
     }
 
     /// <inheritdoc/>
     public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
     {
-        // TODO: add the actual functionality for sampling.
-        throw new System.NotImplementedException();
+        if (this.RulesCache.Expired())
+        {
+            return this.FallbackSampler.ShouldSample(samplingParameters);
+        }
+
+        SamplingRule? matchedRule = this.RulesCache.MatchRule(samplingParameters, this.Resource);
```

Comment:
SamplingParameters are passed as `in` reference. Is it worth to propagate this style for other methods?
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/in-parameter-modifier

Reply:
Sure.

Reply:
done